src/components/project/services.ts

```diff
@@ -0,0 +1,31 @@
+import Vue from 'vue';
+import { AxiosResponse } from 'axios';
+
+import { Project } from './models';
+import { PaginatedResult } from '@/models';
+
+export const getProjects = (): Promise<Project[]> => {
+    return Vue.$axios.get('/project')
+        .then((response: AxiosResponse) => {
+            const data: PaginatedResult<Project> = response.data as PaginatedResult<Project>;
+            return data.results;
+        });
+};
+
+export const getProjectDetails = (id: string): Promise<Project> => {
+    return Vue.$axios.get(`/project/${id}`)
+        .then((response: AxiosResponse) => response.data as Project);
+};
+
+export const createOrUpdateProject = (project: Project): Promise<Project> => {
```

Comment:
Lets break this apart and make create and update their own functions. Your functions should be as small as possible and ideally only do one thing not more. Also the use of ternary operators should be avoid in most cases for a couple reasons: 1) they are harder to read especially since we generally deal with entry level developers. 2) Not they we are performance oriented but they are arguably slower escpecially if you are using them to create functions that should be two different functions. Here is a good stack [post](https://stackoverflow.com/questions/11368948/are-ternary-statements-faster-than-if-then-else-statements-in-javascriptl) about it